tests/testthat/test-build-home-authors.R

```diff
@@ -33,3 +33,20 @@ test_that("Comments in authors info are linkified", {
     "&lt;<a href='https://x.org/'>https://x.org/</a>&gt;)</small>"
   )
 })
+
+test_that("Data authors can accept different filtering", {
+  pkg <- as_pkgdown(test_path("assets/sidebar"))
+  expect_snapshot(data_authors(pkg)$main)
+  expect_snapshot(data_authors(pkg, roles = "cre")$main)
+})
+
+test_that("Text can be added", {
+  pkg <- as_pkgdown(test_path("assets/sidebar-comment"))
+  expect_equal(data_authors_page(pkg)$after, "")
+  expect_equal(data_authors_page(pkg)$before, "")
+
+  pkg$meta$authors$before <- "Dream team:"
```

Comment:
I'd suggest not using markdown here (since that doesn't seem to be the heart of the test) so you can use `expect_equal()` instead of `expect_snapshot()`.